frontend/src/pages/RunDetails.tsx

```diff
@@ -889,6 +889,18 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
         mlmdExecutions,
         namespace,
       });
+
+      // Read optional exeuction id from query parameter. If valid, shows detail of selected node.
+      const paramExecutionId = this.props.match.params[RouteParams.executionId];
+      if (mlmdExecutions) {
+        const selectedExec = mlmdExecutions.find(
+          exec => exec.getId().toString() === paramExecutionId,
+        );
+        if (selectedExec) {
+          const selectedNodeId = ExecutionHelpers.getKfpPod(selectedExec);
+          this.setStateSafe({ selectedNodeDetails: { id: selectedNodeId as string } });
```

Comment:
Instead of casting type, can we make it type safe? E.g. check the other type possibilities are not possible here.

Probably it's better to fix return type of getkdppod

Reply:
Thank you for the suggestion! I changed the return type and make casting type safe.